pex/pip/runtime_patches.py

```diff
@@ -141,7 +141,11 @@ def patch_requires_python():
             orig_check_requires_python = packaging.check_requires_python
 
             def check_requires_python(requires_python, *_args, **_kw):
-                return any(
+                # Ensure any dependency we lock is compatible with the full interpreter range
+                # specified since we have no way to force Pip to backtrack and follow paths for any
+                # divergences. Most (all?) true divergences should be covered by forked environment
+                # markers.
+                return all(
```

Comment:
Checking my understanding: if there is no version that's compatible with the IC range, we now will fail to lock. Whereas before we'd lock w/ a version that might only work with a subset of the ICs. Right?

If so, that's maybe worth a test?